Managers/UInputMan.cpp

```diff
@@ -396,7 +407,20 @@ namespace RTE {
 
 	void UInputMan::TrapMousePos(bool trap, int whichPlayer) {
 		if (whichPlayer == Players::NoPlayer || m_ControlScheme.at(whichPlayer).GetDevice() == InputDevice::DEVICE_MOUSE_KEYB) {
+#ifdef __unix__
+			bool prevTrapStatus = m_TrapMousePos;
+#endif
 			m_TrapMousePos = trap;
+#ifdef __unix__
+			if (m_TrapMousePos != prevTrapStatus) {
+				WarpMouse(_xwin.window_width / 2, _xwin.window_height / 2);
+				int discard;
+				get_mouse_mickeys(&discard, &discard);
```

Comment:
I assume this is deliberate and not accidentally duplicated lines. It seems pretty wonky that you need to do this thrice, but I know you're fighting against ancient allegro so wonkiness is to be expected lol

Reply:
actually this is to get around waiting for the mouse movement events generated by warping the mouse, though not doing it might at most cause the mouse to jump slightly when transitioning between trapped and free mouse on some systems

Reply:
I'm okay with having it, but this would be a case where an inline comment is actually worthwhile, because it's such a non-obvious and weird-looking thing.